### PR TITLE
Add TypeScript decorator support

### PR DESCRIPTION
**Pull request by Timer**
This pull request adds decorator support to TypeScript input only.

### CODE REVIEW

File: packages/next/build/babel/preset.ts [hunk 1]
```diff
@@ -70,12 +70,16 @@ module.exports = (api: any, options: NextBabelPresetOptions = {}): BabelPreset =
         // @babel/plugin-transform-react-jsx-self automatically in development
         development: isDevelopment || isTest,
         ...options['preset-react']
-      }]
+      }],
+      require('@babel/preset-typescript')
```
Comment (timneutkens) [inline packages/next/build/babel/preset.ts]: This looks very wrong tbh, it should only apply for .ts / .tsx

**Comment by timneutkens** [comment]
I don't think we should support decorators for the initial version. Given it's not really an accepted standard yet. Even though TS aims to support it.

File: packages/next/build/babel/preset.ts (hunk 1 above)
Comment (Timer) [reply packages/next/build/babel/preset.ts]: The preset already handles this more proper than we already were. Only the plugin requires the overrides section.

**Comment by Timer** [comment]
@timneutkens decorators are the reason many people choose TypeScript -- not supporting them won't get us far. It's TypeScript's responsibility to migrate people when they switch to the new specification, which probably isn't going to happen until `proposal-decorators` reaches stage 4 / stage 3 🚀.

**Comment by martpie** [comment]
Just my pinch of salt: if decorators are used a lot in Vue for TS or in Angular, I've never seen a React project using decorators (in 2 years).

**Comment by Timer** [comment]
It'd affect MobX users, mainly.

**Pull request closed without merging**